mangaki/mangaki/views.py

```diff
@@ -702,3 +705,32 @@ def error_view(request):
             parameters['origin'] = trope.origin
         return render(request, 'error.html', parameters, status=error_code)
     return error_view
+
+class AnonymousRatingsMixin(object):
+    def get_context_data(self, **kwargs):
+        context = super(AnonymousRatingsMixin, self).get_context_data(**kwargs)
+        ratings = get_anonymous_ratings(self.request.session)
+        works = (
+            Work.objects.filter(id__in=ratings)
+                        .order_by('title')
+                        .group_by_category()
+        )
+        categories = Category.objects.filter(id__in=works).in_bulk()
+        context['ratings'] = [
```

Comment:
It's somewhat complex to read, due to nesting of list comprehensions.
Can we simplify it or extract it into a helper function? Otherwise, some comments would be helpful.

Moreover, it seems like you're building a dict of list, wouldn't be simpler to use a dict comprehension and then use `dict.items` to obtain the tuple form (no idea about performance though.)